File: Scripts/volatile_and_force_compiler_task/factorial/go/factorial.go
package main

func factorial(x uint) uint {
	if x == 0 {
		return 1
	}
	
	return x * factorial(x-1)
}

func executeTask(i int) uint {
	u := uint(i)
	return factorial(u)
}

func main() {
	r := uint(1);

 	for i := 0; i < 100000; i++ {
		r = executeTask(i)
	}
	_=r
}

File: Scripts/volatile_and_force_compiler_task/factorial/c++/factorial.cpp
#include <iostream>

int factorial(int);

int executeTask(int i) {
		return factorial(i);
}

int main() {

volatile int r = 1;
for (int i = 0; i < 100000; ++i) {	
	r =executeTask(i);
}
	return 0; 	
}

int factorial(int number) {
	int temp;

	if(number <= 1) return 1;

	temp = number * factorial(number - 1);
	return temp;
}

